src/ucp/api/ucp.h

```diff
@@ -1172,6 +1213,55 @@ typedef struct ucp_worker_params {
 } ucp_worker_params_t;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP performance endpoint attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the performance of a particular endpoint.
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_perf_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Message size to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    size_t            message_size;
+
+    /**
+     * Local memory type to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t local_memory_type;
+
+    /**
+     * Remote memory type to use for determining performance.
+     * Relevant only for operations that have remote memory access.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t remote_memory_type;
+
+    /**
+     * Operation to report performance for.
+     * This field must be initialized by the caller.
+     */
+    uct_ep_op_t       operation;
```

Comment:
what is required to be set to get estimated_time?  Can i omit some or all input params? What would UCX use then?

Reply:
At the first phase (current implementation), only `message_size` is used for time calculation. `{local,remote}_message_size` and `operation` are here just for future compatibility, but actually they can be safely removed here and added back in a later commit, when their values would really be used.

@yosefe  - shall I explicitly mention that in code comments, or remove these vars at all?

Reply:
i think defaults are worth mentioning for the optional fields

Reply:
Since this is in/out structure, it is really challenging to figure out what is input and what is output.

Reply:
since this is extendable, I would suggest to remove what is not going to be implemented in a near term (before next release) and add as it is gets implemented. This way it also aligns better with your gtest, which tests only the message size.

Reply:
so this means - keep only message size (input) and time (output) for now